### ISSUE TITLE
Play alarm sounds randomly.

### ISSUE THREAD

**Issue by ghost** [BlackyHawky/Clock#98]
**Is your feature request related to a problem? Please describe.**
Play alarm sounds randomly.

**Describe the solution you'd like**
Play alarm sounds randomly.

**Use case**
I like that the alarm sounds change depending on the day.

**Describe alternatives you've considered**
It might help you to implement this feature.
github url
https://github.com/mvarendorff/random-alarm
https://github.com/tal3898/random-alarm

**Additional context**
none.

**Comment by Nilsu11** [comment]
@BlackyHawky for this feature to work we need to add storage permissions because a listener for ending songs/ringtones is non-existent in the ringtone class (same goes for any form of playing which requires changing sounds) . Is this reasonable as a tradeoff for this feature?

**Comment by BlackyHawky** [comment]
@Nilsu11 
> Is this reasonable as a tradeoff for this feature?

What do you mean? (Sorry to ask, I just want to make sure I understand your question.)

**Comment by Nilsu11** [comment]
the tradeoff is the additional storage permission.
So we get the new feature, but also a new permission (for reading music files I think, since the new player would require it). If I (or anyone) were to begin working on the feature, would this permission get in the way of the feature being merged?

**Comment by BlackyHawky** [comment]
I agree that someone could work on this, but I'd like to keep the current ringtone picker interface and not add any other dependencies.

**Comment by Nilsu11** [comment]
@BlackyHawky this [commit](https://github.com/Nilsu11/OpenClock/commit/bc606cb2017469447cec807c6f6a7bb23983ef0f) should implement this feature. note that I only included system alarms.

**Comment by dtwrgj** [comment]
If you access the files the same way as Auxio (v4.0.0-dev5), you may be able to access them without granting music and audio permissions.
https://github.com/OxygenCobalt/Auxio/releases

**Comment by dtwrgj** [comment]
> トレードオフは追加のストレージ権限です。 つまり、新しい機能が得られますが、新しい権限も得られます (新しいプレーヤーには音楽ファイルの読み取り権限が必要だと思います)。私 (または誰か) がこの機能に取り組み始める場合、この権限は機能が統合されるのを妨げますか?

If you access the files the same way as Auxio (v4.0.0-dev5), you may be able to access them without granting music and audio permissions.
https://github.com/OxygenCobalt/Auxio/releases

**Comment by ghost** [comment]
> the tradeoff is the additional storage permission. So we get the new feature, but also a new permission (for reading music files I think, since the new player would require it). If I (or anyone) were to begin working on the feature, would this permission get in the way of the feature being merged?

If you access the files the same way as Auxio (v4.0.0-dev5), you may be able to access them without granting music and audio permissions.
https://github.com/OxygenCobalt/Auxio/releases

**Comment by Storby42** [comment]
+1 on this functionality, would love to have.

**Comment by ghost** [comment]
+1 for this feature.
I think it's best to release apps that require storage permissions and normal apps.
That way, I don't think people who don't need storage permissions will say anything.
＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿＿
Alternatives
If multiple alarms are set at the same time, it is a good idea to select the alarm at random from among them.
That way, I thought there was no need to grant storage privileges.

<img width=200 src="https://github.com/user-attachments/assets/36d4dca6-42c9-4b47-92fe-38e3259ad88e">

However, if you have a lot of music files, changing the alarms one by one can be a pain.

**Comment by KeronCyst** [comment]
I'd love this!

**Comment by BlackyHawky** [comment]
@Nilsu11, @Storby42, @KeronCyst and **others**
I've finally managed to implement random ringtone playback for alarms and timers without causing bugs for alarms, especially when the "At the end of the ringtone" setting is selected.

As expected, the ringtone changes automatically when the alarm is snoozed or dismissed; 

For timers, the change is performed when the timer is restarted.

You can test the attached version below.

[Clock_2.20-beta-debug.zip](https://github.com/user-attachments/files/20260267/Clock_2.20-beta-debug.zip)

Edit: FYI, the next ringtone can also be a custom ringtone.

**Comment by jpgjm** [comment]
We've created a magisk module for those who want to be the first to use custom random calls from users!

Please put your favorite song in the alarm folder.
The alarm path in service.sh may differ depending on the device, so please rewrite it.

[Custom ARAM.zip](https://github.com/user-attachments/files/20261024/Custom.ARAM.zip)

postscript
This module is not needed after the Clock update is officially released, so I'll remove it.

**Comment by BlackyHawky** [comment]
@qw123wh, @Nilsu11, @Storby42, @KeronCyst, @jpgjm and others
What do you think about moving the "Add" button here:

<img width=200 src="https://github.com/user-attachments/assets/944db34a-39b6-4d1d-b9b3-09aab26073fc">

**Comment by qw123wh** [comment]
> [@qw123wh](https://github.com/qw123wh), [@Nilsu11](https://github.com/Nilsu11), [@Storby42](https://github.com/Storby42), [@KeronCyst](https://github.com/KeronCyst), [@jpgjm](https://github.com/jpgjm) and others What do you think about moving the "Add" button here:
> 
> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/139015663/444781546-944db34a-39b6-4d1d-b9b3-09aab26073fc.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDc0OTA1MjAsIm5iZiI6MTc0NzQ5MDIyMCwicGF0aCI6Ii8xMzkwMTU2NjMvNDQ0NzgxNTQ2LTk0NGRiMzRhLTM5YjYtNGQxZC1iOWIzLTA5YWFiMjYwNzNmYy5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxN1QxMzU3MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZGMwYjkzOWUwYzQxM2FlODU4MzExYWZhODM4YjcxYzNkZjA0ODJiYjdjNDliNjM5Y2I2YWFlMmFiZmY4NmE4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.yd_vVz0_18wcZt2H1x82eR8nzqmAhTenZo3Tjp2Wv7s">

Personally I prefer the version as in the apk.

This space at the top could be used in the future if we could implement some sort of media player like in Omniclock, regarding this feature (mediaplayer) I think we should open a new discussion, I found some old commits that should be tested
